packages/apps/plugins/plugin-tree/src/components/Tree/Tree.tsx

```diff
@@ -0,0 +1,321 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import { Square, DotsThreeVertical, X } from '@phosphor-icons/react';
+import React, { type HTMLAttributes, type KeyboardEvent, useEffect, useRef, useState } from 'react';
+
+import { Button, DropdownMenu, Input } from '@dxos/react-ui';
+import { getSize, mx } from '@dxos/react-ui-theme';
+
+import { getNext, getParent, getPrevious, getItems, type Item, getLastDescendent } from './types';
+
+// TODO(burdon): Break/join lines.
+// TODO(burdon): TextObject/MarkdownEditor
+// TODO(burdon): Cut-and-Paste.
+// TODO(burdon): Drag-and-drop.
+// TODO(burdon): Better key nav.
+
+type TreeOptions = Pick<HTMLAttributes<HTMLInputElement>, 'placeholder' | 'spellCheck'> & {
+  checkbox?: boolean;
+};
+
+//
+// Item
+//
+
+type ItemProps = {
+  item: Item;
+  active?: boolean;
+  onFocus?: () => void;
+  onEnter?: (before?: boolean) => void;
+  onDelete?: () => void;
+  onIndent?: (left?: boolean) => void;
+  onNav?: (item: Item, direction?: 'home' | 'end' | 'up' | 'down') => void;
+} & TreeOptions;
+
+const Item = ({
+  item,
+  active,
+  checkbox,
+  placeholder,
+  spellCheck = false,
+  onFocus,
+  onEnter,
+  onDelete,
+  onIndent,
+  onNav,
+}: ItemProps) => {
+  const [focused, setFocused] = useState(false);
```

Comment:
It’s best practices to control focus _outside_ of React, using element `ref`s and `id`s. I’d at least put a to-do here that this should be refactored.

Reply:
I don't understand this point and since I did struggle a bit with this (and also tried to use tabster), perhaps we can discuss offline and then I'll update.